packages/component-catalog-spa/src/App.jsx

```diff
@@ -1,41 +1,113 @@
 // Styles
-import './App.scss';
-import { useEffect, useRef } from 'react';
+import "./App.scss";
+import { useEffect, useRef, useState } from "react";
 // Components
-import '@one-platform/opc-footer/dist/opc-footer';
-import { footer } from './assets/footer';
-import { Breadcrumb, BreadcrumbItem } from '@patternfly/react-core';
+import "@one-platform/opc-footer/dist/opc-footer";
+import { footer } from "./assets/footer";
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  Page,
+  PageSection,
+  Nav,
+  NavList,
+  NavItem,
+  PageSidebar,
+} from "@patternfly/react-core";
 // Pages
-import Main from './pages/Main';
+import Main from "./pages/Main";
+// Routes
+import { Routes, Route, Link } from "react-router-dom";
+import Description from "./Components/Description";
 
-function App() {
+const Home = () => (
+<>
+<main>
+  <h3>Components Catalog</h3>
+  <p>
+    A Unified interface to access components from Chapeaux, Patternfly
+    elements and One Platform components, which allows in increase of
+    collaboration and helps to improve component quality and development & 
+    delivery speed.
+  </p>
+  <Main />
+</main>
+</>);
+
+const breadCrumb = 
+<Breadcrumb>
+  <BreadcrumbItem to="/">Home</BreadcrumbItem>
+  <BreadcrumbItem to="/components-catalogue" isActive>
+    Component Catalog
+  </BreadcrumbItem>
+</Breadcrumb>;
+
+const App = () => {
   const footerRef = useRef();
-  useEffect( () => {
+  useEffect(() => {
     if (footerRef.current) {
       footerRef.current.opcLinkCategories = footer;
     }
-  }, [] );
+  }, []);
+
+  const [navBarItems, setNavbarItems] = useState([]);
+  useEffect(() => {
+    const repositories = [
+      {
+        owner: "1-platform",
+        repo: "op-components",
+        folderName: "packages",
+      },
+      {
+        owner: "patternfly",
+        repo: "patternfly-elements",
+        folderName: "elements",
+      },
+    ];
+    ( async () => {
+      const results = await Promise.all(
+        repositories.map(async (repo) => {
+          const response = await fetch(`https://api.github.com/repos/${repo.owner}/${repo.repo}/contents/${repo.folderName}?ref=master`);
```

Comment:
Its better to create a request config singleton like developer console that contains the URL as an environment variable.

If in future api changes we dont want to replace the code everywhere.

Reply:
Also i feel using then and catch would be better here because first it has multiple request and second  await/async for this will block the browser event loop

Reply:
I agree with the single-config I'll add the changes for it and I think even now it's only being used it one single place but I should be better with a single place